decisiontree/models.py

```diff
@@ -198,6 +199,27 @@ def __str__(self):
         return u"%s : %s --> %s" % (self.current_state, self.answer, self.next_state)
 
 
+class SessionQuerySet(models.query.QuerySet):
+
+    def open(self):
+        return self.exclude(Q(state=None) | Q(canceled=True))
+
+    def complete(self):
+        return self.filter(Q(state=None) | Q(canceled=True))
+
+
+class SessionManager(models.Manager):
```

Comment:
If we're truly OK limiting ourselves to Django 1.7+, then we could use: https://docs.djangoproject.com/en/1.7/topics/db/managers/#creating-manager-with-queryset-methods